scene/gui/color_picker.cpp

```diff
@@ -774,10 +778,76 @@ void ColorPicker::_add_recent_preset_button(int p_size, const Color &p_color) {
 	btn_preset_new->connect("toggled", callable_mp(this, &ColorPicker::_recent_preset_pressed).bind(btn_preset_new));
 }
 
+void ColorPicker::_load_palette() {
+	file_dialog->set_title(RTR("Load Palette"));
+	file_dialog->clear_filters();
+	file_dialog->add_filter("*.cpl");
+
+	file_dialog->set_file_mode(FileDialog::FILE_MODE_OPEN_FILE);
+	file_dialog->set_access(FileDialog::ACCESS_USERDATA);
+	file_dialog->set_current_file("");
+	file_dialog->popup_centered_ratio();
+}
+
+void ColorPicker::_save_palette() {
+	file_dialog->set_title(RTR("Save Palette"));
+	file_dialog->set_current_file("new_palette.cpl");
+	file_dialog->clear_filters();
+	file_dialog->add_filter("*.cpl");
+
+	file_dialog->set_file_mode(FileDialog::FILE_MODE_SAVE_FILE);
+	file_dialog->set_access(FileDialog::ACCESS_USERDATA);
+	file_dialog->popup_centered_ratio();
+}
+
+void ColorPicker::_palette_file_selected(const String &p_path) {
+	switch (file_dialog->get_file_mode()) {
+		case FileDialog::FileMode::FILE_MODE_OPEN_FILE: {
+			Ref<ConfigFile> cf;
+			cf.instantiate();
+			if (cf->load(p_path) == OK) {
+				palette_name->set_text(p_path.get_file().get_basename());
+				palette_name->set_tooltip_text("");
+				preset_cache.clear();
+				presets.clear();
+				PackedColorArray saved_presets = cf->get_value("", "palette", PackedColorArray());
+				for (const Color &saved_preset : saved_presets) {
+					preset_cache.push_back(saved_preset);
+					presets.push_back(saved_preset);
+				}
+			} else {
+				ERR_FAIL_MSG(vformat("Cannot open file to Read."));
+			}
+
+			_update_presets();
+		} break;
+		case FileDialog::FileMode::FILE_MODE_SAVE_FILE: {
+			Ref<ConfigFile> cf;
+			cf.instantiate();
+			cf->set_value("", "palette", get_presets());
+			if (cf->save(p_path) == OK) {
+				palette_name->set_text(p_path.get_file().get_basename());
+				palette_name->set_tooltip_text("");
+			} else {
+				ERR_FAIL_MSG(vformat("Cannot open file to write."));
+			}
+		} break;
```

Comment:
```suggestion
		} break;
		default:
			break;
```
You removed the default as well